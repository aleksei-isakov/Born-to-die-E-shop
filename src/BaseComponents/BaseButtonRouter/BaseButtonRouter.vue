<template>
  <button class="BaseButtonRouter" @click="click">
    <router-link :to="require(path)"><slot></slot></router-link>
  </button>
</template>

<script>
export default {
  name: 'BaseButtonRouter',

  props: {
    path: {
      type: String,
      default: ''
    },

    data() {
      return {};
    },

    methods: {
      click() {
        this.$emit('click', data);
      }
    }
  }
};
</script>
