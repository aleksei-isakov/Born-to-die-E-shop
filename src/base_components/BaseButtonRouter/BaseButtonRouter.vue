<template>
  <button class="base-button-router" @click="sendDataToParent">
    <router-link :to="require(path)"><slot></slot></router-link>
  </button>
</template>

<script>
import Vue from 'vue';
import VueRouter from 'vue-router';

Vue.use(VueRouter);
export default {
  name: 'base-button-router',
  props: {
    path: String
  },
  data() {
    return {};
  },
  methods: {
    sendDataToParent() {
      this.$emit('sendDataToParent', data);
    }
  }
};
</script>

<style lang="scss"></style>
