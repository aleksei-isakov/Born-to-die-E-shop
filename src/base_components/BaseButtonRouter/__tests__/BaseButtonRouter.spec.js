import { shallowMount } from '@vue/test-utils';
import BaseButtonRouter from '../BaseButtonRouter.vue';

describe('BaseButtonRouter.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(BaseButtonRouter, {
      props: {
        path: String
      },
      methods: {
        sendDataToParent() {
          this.$emit('sendDataToParent', data);
        }
      }
    });
    expect(wrapper.find('button')).toBeTruthy();
    expect(wrapper.is(BaseButtonRouter)).toBeTruthy();
  });
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
