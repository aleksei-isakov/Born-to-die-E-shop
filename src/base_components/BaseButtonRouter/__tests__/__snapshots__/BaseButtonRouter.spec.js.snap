// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButtonRouter.vue has the expected html structure 1`] = `
<button
  class="base-button-router"
>
  <router-link-stub
    ariacurrentvalue="page"
    event="click"
    tag="a"
    to="[object Object]"
  />
</button>
`;
